### PR TITLE
GEODE-3469: prevent zero pid from AvailablePid for tests

### PR DESCRIPTION
**Pull request by kirklund**

### CODE REVIEW

File: geode-core/src/test/java/org/apache/geode/internal/process/lang/AvailablePidTest.java [hunk 1]
```diff
@@ -100,7 +105,53 @@ public void findAvailablePidsShouldReturnNoDuplicatedPids() throws Exception {
...
+    availablePid = new AvailablePid(new AvailablePid.Bounds(1, 3));
+
+    await().atMost(10, SECONDS).until(() -> assertThat(availablePid.random()).isEqualTo(1));
+  }
+
+  @Test
+  public void randomUpperBoundIsInclusive() throws Exception {
+    availablePid = new AvailablePid(new AvailablePid.Bounds(1, 3));
+
+    await().atMost(10, SECONDS).until(() -> assertThat(availablePid.random()).isEqualTo(3));
```
Comment (jaredjstewart) [inline geode-core/src/test/java/org/apache/geode/internal/process/lang/AvailablePidTest.java]: This is a clever way to test that lower and upper bounds are potential output values!

File: geode-core/src/test/java/org/apache/geode/internal/process/lang/AvailablePidTest.java [hunk 2]
```diff
@@ -100,7 +105,53 @@ public void findAvailablePidsShouldReturnNoDuplicatedPids() throws Exception {
...
   }
 
-  private void assertThatNoPidIsDuplicated(int[] pids) {
+  @Test(timeout = DEFAULT_TIMEOUT_MILLIS)
+  public void findAvailablePidShouldReturnGreaterThanOrEqualToLowerBound() throws Exception {
+    availablePid = new AvailablePid(new AvailablePid.Bounds(1, 3));
+
+    int pid = availablePid.findAvailablePid();
+
+    assertThat(pid).isGreaterThanOrEqualTo(1);
```
Comment (jaredjstewart) [inline geode-core/src/test/java/org/apache/geode/internal/process/lang/AvailablePidTest.java]: I wonder if trying random values for some time period (like the awaitility tests below) might be more robust:

```
availablePid = new AvailablePid(new AvailablePid.Bounds(1, 3));

com.google.common.base.Stopwatch stopwatch = Stopwatch.createStarted();
while (stopwatch.elapsed(TimeUnit.SECONDS) < 10) {
  int pid = availablePid.findAvailablePid();
  assertThat(pid).isLessThanOrEqualTo(3);
}

File: geode-core/src/test/java/org/apache/geode/internal/process/lang/AvailablePid.java [hunk 3]
```diff
@@ -30,37 +30,88 @@
...
   private final Random random;
   private final int timeoutMillis;
 
   /**
-   * Construct with no seed and default timeout of 1 minute.
+   * Construct with:
+   * <ul>
+   * <li>default {@link Bounds} of {@link #DEFAULT_LOWER_BOUND} (inclusive) and
+   * {@link #DEFAULT_UPPER_BOUND} (inclusive)
+   * <li>Random with no see
```
Comment (jaredjstewart) [inline geode-core/src/test/java/org/apache/geode/internal/process/lang/AvailablePid.java]: I think you meant "no see**d**"

Comment (kirklund) [inline geode-core/src/test/java/org/apache/geode/internal/process/lang/AvailablePid.java]: I no see your point ;)

**Comment by kirklund** [comment]
Already merged in.

**Pull request closed without merging**